tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/Lambda/LambdaCommon.cs

```diff
@@ -57,14 +43,16 @@ internal static Scope SendStartInvocation(ILambdaExtensionRequest requestBuilder
         WriteRequestPayload(request, data);
         WriteRequestHeaders(request, context);
         var response = (HttpWebResponse)request.GetResponse();
-        var traceId = response.Headers.Get(HttpHeaderNames.TraceId);
-        var samplingPriority = response.Headers.Get(HttpHeaderNames.SamplingPriority);
-        if (ValidateOkStatus(response))
+
+        // response.Headers is a WebHeaderCollection, which is not convertable to IHeadersCollection, so we have to make it into a dictionary.
+        var myHeaders = response.Headers.AllKeys.ToDictionary(k => k, k => response.Headers.Get(k));
```

Comment:
A few things here:

We generally avoid LINQ for performance reasons, but we don't need to create and populate a `Dictionary<string,string` anyway because...

If you need a `IHeadersCollection`, we have a wrapper for `WebHeaderCollection` which implements `IHeadersCollection`, imaginatively called `WebHeadersCollection`:
https://github.com/DataDog/dd-trace-dotnet/blob/master/tracer/src/Datadog.Trace/Headers/WebHeadersCollection.cs

But wait, don't use `WebHeadersCollection` either 😅  We have been moving away from using `IHeadersCollection`. The only use left that I could find was in the WCF integration [here](https://github.com/DataDog/dd-trace-dotnet/blob/dce153d44f53d98a5fb12f22a77f97469b8a0f7b/tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Wcf/WcfCommon.cs#L63). We now have overloads of `SpanContextPropagator.Extract` which [take a getter function](https://github.com/DataDog/dd-trace-dotnet/blob/dce153d44f53d98a5fb12f22a77f97469b8a0f7b/tracer/src/Datadog.Trace/Propagators/SpanContextPropagator.cs#L153) instead of `IHeadersCollection`.

You should be able to do something like:
```csharp
var response = (HttpWebResponse)request.GetResponse();
var headers = response.Headers; 
...
var context = SpanContextPropagator.Instance.Extract(headers, (h, name) => h.GetValues(name));
```

Note that this overload of `SpanContextPropagator.Extract` expects the getter to return `IEnumerable<string?>`, so we're using `WebHeaderCollection.GetValues(string name)` which returns a `string?[]`.

Reply:
It took me an embarrassing amount of time to figure out how to implement this vs. how simple the implementation was. Thank you!

Reply:
> It took me an embarrassing amount of time to figure out how to implement this

Always feel free to ping us on Slack, happy to help!